modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C

```diff
@@ -77,58 +77,90 @@ ComputeReturnMappingStress::updateQpStress(RankTwoTensor & strain_increment,
       << std::endl;
   }
 
-  // compute trial stress
-  RankTwoTensor stress_last_iteration = _elasticity_tensor[_qp] * (strain_increment + _elastic_strain_old[_qp]);
-
   RankTwoTensor inelastic_strain_increment, elastic_strain_increment;
```

Comment:
I'd get rid of the `inelastic_strain_increment` variable. You can just pass the inelastic strain increment for the current model in to `updateStress()`.